app/src/ui/lib/git-email-not-found-warning.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react'
+import { Account } from '../../models/account'
+import { LinkButton } from './link-button'
+import { getDotComAPIEndpoint } from '../../lib/api'
+
+interface IGitEmailNotFoundWarningProps {
+  /** The account the commit should be attributed to. */
+  readonly accounts: ReadonlyArray<Account>
+
+  /** The email address used in the commit author info. */
+  readonly email: string
+}
+
+/**
+ * A component which just displays a warning to the user if their git config
+ * email doesn't match any of the emails in their GitHub (Enterprise) account.
+ */
+export class GitEmailNotFoundWarning extends React.Component<
+  IGitEmailNotFoundWarningProps
+> {
+  private get accountEmails(): ReadonlyArray<string> {
+    // Merge email addresses from all accounts into an array
+    return this.props.accounts.reduce<ReadonlyArray<string>>(
+      (previousValue, currentValue) => {
+        return previousValue.concat(currentValue.emails.map(e => e.email))
+      },
+      []
+    )
+  }
+
+  public render() {
+    if (this.accountEmails.includes(this.props.email)) {
+      return null
+    }
+
+    return this.props.accounts.length === 1
+      ? this.renderForSingleAccount(this.props.accounts[0])
+      : this.renderForMultipleAccounts()
+  }
+
+  private renderForSingleAccount(account: Account) {
+    const accountType =
+      account.endpoint === getDotComAPIEndpoint()
+        ? 'GitHub'
```

Comment:
Good point! I'm not sure what's the correct thing here, because I see we use `GitHub.com` only when it's shown along with `GitHub Enterprise`, and then we use just `GitHub` when it's shown independently 🤔

Reply:
Ah! Well if there is already a convention of doing so then I would think to just leave it as is.